doc/classes/Geometry2D.xml

```diff
@@ -125,8 +125,28 @@
 			<param index="2" name="from_b" type="Vector2" />
 			<param index="3" name="dir_b" type="Vector2" />
 			<description>
-				Checks if the two lines ([param from_a], [param dir_a]) and ([param from_b], [param dir_b]) intersect. If yes, return the point of intersection as [Vector2]. If no intersection takes place, returns [code]null[/code].
-				[b]Note:[/b] The lines are specified using direction vectors, not end points.
+				Checks if the two lines ([param from_a], [param dir_a]) and ([param from_b], [param dir_b]) intersect. Returns a [Vector2], or [code]null[/code] if the lines are parallel.
+				[code]from[/code] and [code]dir[/code] are **not** endpoints of a line segment or ray but the slope ([code]dir[/code]) and a known point ([code]from[/code]) on that line.
+				[codeblocks]
+				[gdscript]
+				var from_a = Vector2.ZERO
+				var dir_a = Vector2.RIGHT
+				var from_b = Vector2.DOWN    # y == 1 in 2D
+				
+				var intersect_right = Geometry2D.line_intersects_line(from_a, dir_a, from_b, Vector2(1, -1))     # returns Vector2(1, 0)
+				var intersect_left  = Geometry2D.line_intersects_line(from_a, dir_a, from_b, Vector2(-1, -1))    # returns Vector2(-1, 0)
+				var intersect_none  = Geometry2D.line_intersects_line(from_a, dir_a, from_b, Vector2.RIGHT)      # returns null
```

Comment:
These lines are too long and will not display as nicely as you imagine. I would suggest putting the comment on a new line.

For this kind of example, what I have more often seen is "Variable is value", over "Returns value". The latter is completely fine, too, but it could be done only when not actually assigning the return value. It's a very opinionated suggestion, though.
```suggestion
				# intersect_right is Vector2(1, 0)
				var intersect_right = Geometry2D.line_intersects_line(from_a, dir_a, from_b, Vector2(1, -1))
				# intersect_left is Vector2(-1, 0)
				var intersect_left  = Geometry2D.line_intersects_line(from_a, dir_a, from_b, Vector2(-1, -1))
				# intersect_none is null
				var intersect_none  = Geometry2D.line_intersects_line(from_a, dir_a, from_b, Vector2.RIGHT)
```

Reply:
I went with the latter in the interests of further shortening the lines. I think it's more readable and it doesn't really do much for readability to assign the results and add an example comment